src/datanode/src/heartbeat/handler/downgrade_region.rs

```diff
@@ -89,33 +96,35 @@ impl HandlerContext {
                 return self.downgrade_to_follower_gracefully(region_id).await;
             };
 
-            if reject_write {
-                // Sets region to downgrading, the downgrading region will reject all write requests.
-                match self
-                    .region_server
-                    .set_region_role_state_gracefully(
-                        region_id,
-                        SettableRegionRoleState::DowngradingLeader,
-                    )
-                    .await
-                {
-                    Ok(SetRegionRoleStateResponse::Success { .. }) => {}
-                    Ok(SetRegionRoleStateResponse::NotFound) => {
-                        warn!("Region: {region_id} is not found");
-                        return Some(InstructionReply::DowngradeRegion(DowngradeRegionReply {
-                            last_entry_id: None,
-                            exists: false,
-                            error: None,
-                        }));
-                    }
-                    Err(err) => {
-                        warn!(err; "Failed to convert region to downgrading leader");
-                        return Some(InstructionReply::DowngradeRegion(DowngradeRegionReply {
-                            last_entry_id: None,
-                            exists: true,
-                            error: Some(format!("{err:?}")),
-                        }));
-                    }
+            // Sets region to downgrading,
+            // the downgrading region will reject all write requests.
+            // However, the downgrading region will still accept read, flush requests.
+            match self
+                .region_server
+                .set_region_role_state_gracefully(
+                    region_id,
+                    SettableRegionRoleState::DowngradingLeader,
+                )
+                .await
+            {
+                Ok(SetRegionRoleStateResponse::Success { .. }) => {}
+                Ok(SetRegionRoleStateResponse::NotFound) => {
+                    warn!("Region: {region_id} is not found");
+                    return Some(InstructionReply::DowngradeRegion(DowngradeRegionReply {
+                        last_entry_id: None,
+                        metadata_last_entry_id: None,
+                        exists: false,
+                        error: None,
+                    }));
+                }
+                Err(err) => {
+                    warn!(err; "Failed to convert region to downgrading leader");
```

Comment:
The error must be logged as an error.
```suggestion
                    error!(err; "Failed to convert region to downgrading leader");
```